base/div.jl

```diff
@@ -0,0 +1,188 @@
+
+- if `r == RoundNearest`, then the result is exact, and in the interval
+  ``[-|y|/2, |y|/2]``. See also [`RoundNearest`](@ref).
+
+- if `r == RoundToZero` (default), then the result is exact, and in the interval
+  ``[0, |y|)`` if `x` is positive, or ``(-|y|, 0]`` otherwise. See also [`RoundToZero`](@ref).
+
+- if `r == RoundDown`, then the result is in the interval ``[0, y)`` if `y` is positive, or
+  ``(y, 0]`` otherwise. The result may not be exact if `x` and `y` have different signs, and
+  `abs(x) < abs(y)`. See also[`RoundDown`](@ref).
+
+- if `r == RoundUp`, then the result is in the interval `(-y,0]` if `y` is positive, or
+  `[0,-y)` otherwise. The result may not be exact if `x` and `y` have the same sign, and
+  `abs(x) < abs(y)`. See also [`RoundUp`](@ref).
+
+"""
+rem(x, y, r::RoundingMode)
+
+# TODO: Make these primitive and have the two-argument version call these
+rem(x, y, ::RoundingMode{:ToZero}) = rem(x,y)
+rem(x, y, ::RoundingMode{:Down}) = mod(x,y)
+rem(x, y, ::RoundingMode{:Up}) = mod(x,-y)
+
+"""
+    fld(x, y)
+
+Largest integer less than or equal to `x/y`. Equivalent to `div(x, y, RoundDown)`.
+
+See also: [`div`](@ref)
+
+# Examples
+```jldoctest
+julia> fld(7.3,5.5)
+1.0
+```
+"""
+fld(a, b) = div(a, b, RoundDown)
+
+"""
+    cld(x, y)
+
+Smallest integer larger than or equal to `x/y`. Equivalent to `div(x, y, RoundUp)`.
+
+See also: [`div`](@ref)
+
+# Examples
+```jldoctest
+julia> cld(5.5,2.2)
+3.0
+```
+"""
+cld(a, b) = div(a, b, RoundUp)
+
+# divmod
+divrem(a, b, r::RoundingMode) = (div(a, b, r), rem(a, b, r))
+
+"""
+    fldmod(x, y)
+
+The floored quotient and modulus after division. A convenience wrapper for
+`divrem(x, y, RoundDown)`. Equivalent to `(fld(x,y), mod(x,y))`.
+"""
+fldmod(x,y) = divrem(a, b, RoundDown)
+
+# We definite generic rounding methods for other rounding modes in terms of
+# RoundToZero.
+function div(x::Signed, y::Unsigned, ::typeof(RoundDown))
+    (q, r) = divrem(x, y)
+    q - (signbit(x) & (r != 0))
+end
+function div(x::Unsigned, y::Signed, ::typeof(RoundDown))
+    (q, r) = divrem(x, y)
+    q - (signbit(y) & (r != 0))
+end
+
+function div(x::Signed, y::Unsigned, ::typeof(RoundUp))
+    (q, r) = divrem(x, y)
+    q + (!signbit(x) & (r != 0))
+end
+function div(x::Unsigned, y::Signed, ::typeof(RoundUp))
+    (q, r) = divrem(x, y)
+    q + (!signbit(y) & (r != 0))
+end
+
+function div(x::Integer, y::Integer, rnd::Union{typeof(RoundNearest),
+                                              typeof(RoundNearestTiesAway),
+                                              typeof(RoundNearestTiesUp)})
+    (q, r) = divrem(x, y)
+    halfy = copysign(y >> 1, r)
+    c = cmp(r, halfy)
+    c == -1 && return q
+    c == 1 && return q + copysign(1, q)
+    if rnd == RoundNearest
+        return iseven(q) ? q : q + copysign(1, q)
+    elseif rnd == RoundNearestTiesAway
+        return q + copysign(1, q)
+    else
+        @assert rnd == RoundNearestTiesUp
+        return sign(q) == -1 ? q : q + 1
```

Comment:
I don't think it handles negative odd values of `y` correctly, e.g. `divrem(5, -3, RoundNearestTiesUp)`.